packages/loopring_v3/contracts/impl/BlockVerifier.sol

```diff
@@ -41,54 +48,92 @@ contract BlockVerifier is IBlockVerifier, Claimable
         external
         onlyOwner
     {
-        // While vk[0] could be 0, we don't allow it so we can use this to easily check
-        // if the verification key is set
-        require(vk[0] != 0, "INVALID_DATA");
         bool dataAvailability = needsDataAvailability(blockType, onchainDataAvailability);
         require(dataAvailability == onchainDataAvailability, "NO_DATA_AVAILABILITY_NEEDED");
+        Circuit storage circuit = circuits[onchainDataAvailability][blockType][blockSize][blockVersion];
+        require(circuit.registered == false, "ALREADY_REGISTERED");
```

Comment:
It seems:
1) once a circuit is disabled, it cannot be re-enabled. 
2) once a circuit is registered, we cannot re-register it again with a new set of new keys. 

Both seem OK to me.

Reply:
That's correct.